src/internet_identity/src/main.rs

```diff
@@ -114,17 +124,23 @@ async fn add_tentative_device(
 #[update]
 fn verify_tentative_device(
     anchor_number: AnchorNumber,
-    user_verification_code: DeviceVerificationCode,
+    user_confirmation_code: DeviceConfirmationCode,
 ) -> VerifyTentativeDeviceResponse {
-    let result = anchor_operation_with_authz_check(anchor_number, |anchor| {
-        tentative_device_registration::verify_tentative_device(
-            anchor,
-            anchor_number,
-            user_verification_code,
-        )
-    });
-    match result {
-        Ok(()) => VerifyTentativeDeviceResponse::Verified,
+    let (mut anchor, authorization_key) = check_authorization(anchor_number)
+        .unwrap_or_else(|err| trap(&format!("{} could not be authenticated.", err.principal)));
+    match tentative_device_registration::confirm_tentative_device(
+        anchor_number,
+        user_confirmation_code,
+    ) {
+        Ok(confirmed_device) => {
```

Comment:
Could we rename `Ok(maybe_confirmed_device)` to make it clear that there is an option within the `Ok()`.

Or mabye even `Ok(Some(confirmed_device))` and `Ok(None)` and you don't need the nested conditional.